Source/santad/Logs/EndpointSecurity/Serializers/BasicString.mm

```diff
@@ -168,6 +179,64 @@ static inline void AppendUserGroup(std::string &str, const audit_token_t &tok,
   str.append(group.has_value() ? group->get()->c_str() : "(null)");
 }
 
+static inline void AppendEventUser(std::string &str, const es_string_token_t &user,
+                                   std::optional<uid_t> uid) {
+  if (user.length > 0) {
+    str.append("|event_user=");
+    str.append(user.data);
+  }
+
+  if (uid.has_value()) {
+    str.append("|event_uid=");
+    str.append(std::to_string(uid.value()));
+  }
+}
+
+static inline void AppendInstigator(std::string &str, const EnrichedEventType &event) {
+  AppendProcess(str, event->process);
+  AppendUserGroup(str, event->process->audit_token, event.instigator().real_user(),
+                  event.instigator().real_group());
+}
+
+static inline void AppendGraphicalSession(std::string &str, es_graphical_session_id_t session_id) {
+  str.append("|graphical_session_id=");
+  str.append(std::to_string(session_id));
+}
+
+static inline void AppendSocketAddress(std::string &str, es_address_type_t type,
+                                       es_string_token_t addr) {
+  str.append("|address_type=");
+  switch (type) {
+    case ES_ADDRESS_TYPE_NONE: str.append("none"); break;
+    case ES_ADDRESS_TYPE_IPV4: str.append("ipv4"); break;
+    case ES_ADDRESS_TYPE_IPV6: str.append("ipv6"); break;
+    case ES_ADDRESS_TYPE_NAMED_SOCKET: str.append("named_socket"); break;
```

Comment:
WDYT about using `unix` or `uds` here instead of `named_socket`?

Reply:
I'm fine with using the full `unix_domain_socket` name, will change.

Reply:
ES calls it `named_socket`, I think matching that makes the most sense.
https://developer.apple.com/documentation/endpointsecurity/es_address_type_t?language=objc

If we were to deviate, maybe `local_socket`?
From `man 2 socket`: `PF_LOCAL        Host-internal protocols, formerly called PF_UNIX`

Reply:
I agree staying close to the ES data here makes sense. Though it'll require some documentation.

Reply:
Keeping the name to match closely with ES.